File: flower/src/main/java/com/kharitonov/flower/exceptions/CustomExceptionHandler.java
package com.kharitonov.flower.exceptions;

import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.context.request.WebRequest;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@Slf4j
@RestControllerAdvice
public class CustomExceptionHandler extends ResponseEntityExceptionHandler {

    @ExceptionHandler
    public ResponseEntity<Object> handler(Exception exception, WebRequest request) throws Exception {
        log.error("Exception during execution of application", exception);
        return handleException(exception, request);
    }
}

File: src/main/java/by/hghgrtut/shop/exceptions/CustomExceptionHandler.kt
package by.hghgrtut.shop.exceptions

import lombok.extern.slf4j.Slf4j

@Slf4j
@RestControllerAdvice
class CustomExceptionHandler : ResponseEntityExceptionHandler() {

    @ExceptionHandler
    @Throws(Exception::class)
    fun handler(exception: Exception?, request: WebRequest?): ResponseEntity<Object> {
        log.error("Exception during execution of application", exception)
        return handleException(exception, request)
    }
}